libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -2950,6 +2950,24 @@ void GCS_MAVLINK::handle_statustext(const mavlink_message_t &msg) const
 }
 
 
+/*
+  handle logging of named values from mavlink.
+ */
+void GCS_MAVLINK::handle_named_value(const mavlink_message_t &msg) const
+{
+    auto *logger = AP_Logger::get_singleton();
+    if (logger == nullptr) {
+        return;
+    }
+    mavlink_named_value_float_t p;
+    mavlink_msg_named_value_float_decode(&msg, &p);
+    logger->Write("NVAL", "TimeUS,TimeBootMS,Name,Value", "ss#-", "FC--", "QINf",
+                  AP_HAL::micros64(),
+                  p.time_boot_ms,
+                  p.name,
```

Comment:
I don't think `p` is guaranteed null-terminated here, thus the `memcpy` shenanigans in `handle_statustext`.

Reply:
fixed